### PR TITLE
Add docstrings

### PR DESCRIPTION
**Pull request by lrennels**

### CODE REVIEW

File: src/core/defs.jl [hunk 1]
```diff
@@ -578,7 +578,10 @@ unit(obj::ParameterDefReference) = parameter(obj).unit
     variable_dimensions(obj::AbstractCompositeComponentDef, comp_path::ComponentPath, var_name::Symbol)
 
 Return the names of the dimensions of variable `var_name` exposed in the composite 
-component definition indicated by`obj` along the component path `comp_path`.
+component definition indicated by`obj` along the component path `comp_path`. The 
+`comp_path`is of type `Mimi.ComponentPath` with the single arugment being a named
```
Comment (ckingdon95) [inline src/core/defs.jl]: "argument" typo

Comment (lrennels) [reply src/core/defs.jl]: @ckingdon95 fixed!

**Comment by ckingdon95** [comment]
Do we need this PR now? it looks like these things were also added in the other one?

**Comment by lrennels** [comment]
Ah true I should edit them in that one too.  The other one was stalling for so long that I wanted to get this part done, since it was breaking the docs build.

**Comment by ckingdon95** [comment]
I meant the "variable_dimensions" one. I think that one can be merged now (it's passing) and then we don't need this one? Cause you also added these docstrings to that one?

**Comment by lrennels** [comment]
Yep you're right I just changed that, so the only thing that this one does now is add docstrings for `TimestepValue` and `TimestepIndex` which we do need, because they are in the reference.md file but missing docstrings.

**Comment by lrennels** [comment]
Oh they're both in that PR!  Ok yes you're right let me fix that right now.

**Pull request closed without merging**